src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -75,20 +75,24 @@ internal override async ContractTask OnPersistAsync(ApplicationEngine engine)
                         var payer = tx.Signers[1];
                         // Don't need to seal because Deposit is a fixed-sized interoperable, hence always can be serialized.
                         var balance = engine.SnapshotCache.GetAndChange(CreateStorageKey(Prefix_Deposit, payer.Account))?.GetInteroperable<Deposit>();
-                        balance.Amount -= tx.SystemFee + tx.NetworkFee;
-                        if (balance.Amount.Sign == 0) RemoveDepositFor(engine.SnapshotCache, payer.Account);
+                        if (balance != null)
+                        {
+                            balance.Amount -= tx.SystemFee + tx.NetworkFee;
+                            if (balance.Amount.Sign == 0) RemoveDepositFor(engine.SnapshotCache, payer.Account);
+                        }
+                        // TODO: Else?
```

Comment:
Roman agrees with proposed solution. So else just continue the cycle iteration.

No balance - nothing to withdraw - skip the account, there's nothing we can do about it because this transaction is already accepted by consensus.